### ISSUE TITLE
Aliasing QueryProxy without losing scope

### ISSUE THREAD

**Issue by ernestoe** [neo4jrb/neo4j#1278]
I have the following ActiveNode class
````
[1] pry(main)> DD::Error.count
 CYPHER 17ms MATCH (n:`DD::Error`) RETURN count(n) AS count 
=> 30990
`````
a scope defined in order to filter the class:
````
  scope :active, -> { where(status: 'active') }
````
so that:
````
[4] pry(main)> DD::Error.active.count
 CYPHER 34ms MATCH (result_dderror:`DD::Error`) WHERE (result_dderror.status = {result_dderror_status}) RETURN count(result_dderror) AS result_dderror | {:result_dderror_status=>"active"}
=> 2646
[5] pry(main)> my_qp = DD::Error.active
=> #<QueryProxy  CYPHER: "MATCH (result_dderror:`DD::Error`) WHERE (result_dderror.status = {result_dderror_status})">
[6] pry(main)> my_qp.count
 CYPHER 33ms MATCH (result_dderror:`DD::Error`) WHERE (result_dderror.status = {result_dderror_status}) RETURN count(result_dderror) AS result_dderror | {:result_dderror_status=>"active"}
=> 2646
````
I need to alias the QueryProxy in order to do some dynamic filtering later on. However it doesn't seem to work.
````
[9] pry(main)> my_qp.as(:a).count
 DD::Error 13ms MATCH (a:`DD::Error`) RETURN count(a) AS a 
=> 30990
````
However using the following combination works properly:
````
[11] pry(main)> my_qp.query_as(:a).proxy_as(DD::Error, :a).count
 CYPHER 41ms MATCH (a:`DD::Error`) WHERE (a.status = {a_status}) RETURN count(a) AS a | {:a_status=>"active"}
=> 2646
````

Is this expected behaviour?
### Runtime information:

Neo4j database version:
`neo4j` gem version: 7.2.0
`neo4j-core` gem version: 6.1.4

**Comment by cheerfulstoic** [comment]
This is definitely a bug, thanks for pointing it out!

**Comment by cheerfulstoic** [comment]
@ernestoe This should be fixed in `8.0.0.alpha.12` (thanks again @ProGM )
